packages/template-retail-react-app/app/pages/checkout/partials/contact-info.jsx

```diff
@@ -96,6 +96,12 @@ const ContactInfo = () => {
         authModal.onOpen()
     }
 
+    useEffect(() => {
+        if (!showPasswordField) {
+            form.unregister('password')
+        }
+    }, [showPasswordField])
```

Comment:
Could this be done in `togglePasswordField`?

like
```
const togglePasswordField = () => {
        if (error) {
            setError(null)
        }
        setShowPasswordField(!showPasswordField)
        if (!showPasswordField) {
           form.unregister('password')
        }
    }
```

Reply:
also, do we need to re-register password field when `showPasswordField` is toggled to true?

Reply:
We don't need to re register. It registers automatically since the component mounts and unmounts instead of simple show/hide.

Reply:
My original solution was to implenent it inside `togglePasswordField` but I believe setState is async. So the if condition probably executes before setState is complete and the if never evaluates to true.

So I had to move it into a `useEffect` outside of the function to guarantee state change is caught.